addons/mail/static/src/models/emoji_grid_view.js

```diff
@@ -185,6 +185,16 @@ registerModel({
         scrollRecomputeCount: attr({
             default: 0,
         }),
+        searchNoContentView: one('EmojiGridNoSearchContentView', {
+            compute() {
+                if (this.emojiPickerViewOwner.emojiSearchBarView.currentSearch !== "" && this.rows.length === 0) {
```

Comment:
There are no rows because there are no search results. The NoContentView is ultimately based on the search results. I feel like you have "indirected" the logic of these models too much.

Reply:
- `currentSearch !== ""` means in "search mode"
- `rows.length === 0` means no rows to display

The "search no content view" is a record that exists when there are no row in "search mode".

Reply:
I think of "rows" as a grid's concept, so if we agree to say that `EmojiGridView` is not really a grid but rather the container below the search bar, I think it shouldn't have the concept of rows.